main-aws.tf

```diff
@@ -102,6 +102,32 @@ resource "aws_s3_bucket" "data" {
     git_repo             = "terragoat"
     yor_trace            = "0874007d-903a-4b4c-945f-c9c233e13243"
   })
+
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm     = "aws:kms"
+        kms_master_key_id = "<master_kms_key_id>"
+      }
+    }
+  }
+
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+
+  versioning {
+    enabled = true
+  }
+
```

Comment:
**Tenable.cs remediation**
 
**[HIGH]: Ensure bucket policy is enforced with least privileges for all AWS S3 buckets**
 
**Vulnerability**
S3 bucket Access is allowed to all AWS Account Users.
 
**Remediation**
Make sure to note Amazon S3 bucket access control lists (ACLs) that provide read, write, or full-access to 'Everyone'. It is recommended to provide access to S3 buckets to only authenticated users instead.

source: tenablecs-